test/end2end_test.go

```diff
@@ -1595,6 +1595,61 @@ func testExceedMaxStreamsLimit(t *testing.T, e env) {
 	}
 }
 
+func TestStreamsQuotaRecovery(t *testing.T) {
+	defer leakCheck(t)()
+	for _, e := range listTestEnv() {
+		testStreamsQuotaRecovery(t, e)
+	}
+}
+
+func testStreamsQuotaRecovery(t *testing.T, e env) {
+	te := newTest(t, e)
```

Comment:
The non-deterministic nature of the bug is key.  If you can't verify that you address the issue in the unlucky case then you can regress at a later point in time and you would not know it.

Again, not trying to block just commenting on the contents of the review.